File: src/1518.water-bottles.1642/solution.ts
function numWaterBottles (numBottles: number, numExchange: number): number {
    let result = 0;
    let empty = 0;
    while (numBottles > 0) {
        result += numBottles;
        empty += numBottles;
        const rest = empty % numExchange;
        numBottles = (empty - rest) / numExchange;
        empty = rest;
    }
    return result;
}

File: src/1518.water-bottles.1642/solution.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int result = numBottles;
        int empty = numBottles;
        while(empty>=numExchange){
            int newBottle = empty/numExchange;
            int rest = empty%numExchange;
            result += newBottle;
            empty = rest+newBottle;
        }
        return result;
    }
}